metrics-exporter-http/src/lib.rs

```diff
@@ -45,33 +41,24 @@ where
         }
     }
 
-    /// Run the exporter on the current thread.
-    ///
-    /// This starts an HTTP server on the `address` the exporter was originally configured with,
-    /// responding to any request with the output of the configured observer.
-    pub fn run(self) {
-        let server = self.into_future();
-        hyper_run(server);
-    }
-
     /// Converts this exporter into a future which can be driven externally.
     ///
     /// This starts an HTTP server on the `address` the exporter was originally configured with,
     /// responding to any request with the output of the configured observer.
-    pub fn into_future(self) -> impl Future<Item = (), Error = ()> {
+    pub async fn into_future(self) -> hyper::error::Result<()> {
         let controller = self.controller;
         let builder = self.builder;
         let address = self.address;
 
-        build_hyper_server(controller, builder, address)
+        build_hyper_server(controller, builder, address).await
     }
 }
 
-fn build_hyper_server<C, B>(
+async fn build_hyper_server<C, B>(
```

Comment:
We should collapse this into `into_future` because it's only used there now.

Reply:
Okay, done. But now I notice the whole `HttpExporter` struct is looking a little silly, because calling `into_future()` is literally the only thing you can do with it. Perhaps it should just be replaced with an async fn that returns the future directly?

Reply:
Yeah, let me think on this a little bit... I had always sort of intended this this crate would be equally usable in both sync and async code.  Gotta figure out what the right API looks like here.

Reply:
I suppose I could add back the `run()` method, and have it set up a basic_scheduler tokio runtime and then block_on the future? Not sure how appealing that's going to be, I think most users' attitude is going to be that if their program is going to have a tokio runtime in it, they don't want it abstracted away.